esp/platform/espcfg.cpp

```diff
@@ -272,10 +276,45 @@ CEspConfig::CEspConfig(IProperties* inputs, IPropertyTree* envpt, IPropertyTree*
     {
         DBGLOG("ESP process name [%s]", m_process.str());
 
+        setDetachedFromDali(false);
+        try
+        {
+            StringBuffer procDirectory;
+            m_cfg->getProp("@directory", procDirectory);
+            m_esdlStateFilesLocation.setf("%s%c%s-AttachState.xml",procDirectory.str(), PATHSEPCHAR, m_process.str());
+
+            try
+            {
+                Owned<IPTree> espProcAttachState = createPTreeFromXMLFile(m_esdlStateFilesLocation);
+                if (espProcAttachState)
+                {
+                    const char * restoredBindingTS = espProcAttachState->queryProp("@StateSaveTimems");
+                    setDetachedFromDali(!(espProcAttachState->getPropBool("@attached", true)));
+                    bool subscribed = espProcAttachState->getPropBool("@subscribed", true);
+                    ESPLOG(LogMin, "ESP is %s attached to DALI", m_detachedFromDali? "NOT" : "");
+                }
+                else
+                {
+                    ESPLOG(LogMin, "Failed to load DESDL binding state from local file: '%s' for ESP binding %s", m_esdlStateFilesLocation.str(), m_process.str());
+                }
+            }
+            catch (...)
+            {
+                ESPLOG(LogMin, "Failed to load DESDL binding state from local file: '%s' for ESP binding %s", m_esdlStateFilesLocation.str(), m_process.str());
+            }
+        }
+        catch (...)
+        {
+            ESPLOG(LogMin, "Failed to load DESDL binding state from local file: '%s' for ESP binding %s", m_esdlStateFilesLocation.str(), m_process.str());
+        }
+
         IPropertyTreeIterator *pt_iter = NULL;
-        StringBuffer daliservers;
-        if (m_cfg->getProp("@daliServers", daliservers))
-            initDali(daliservers.str());
+        if(!isDetachedFromDali())
+        {
+            StringBuffer daliservers;
+            if (m_cfg->getProp("@daliServers", daliservers))
```

Comment:
Haven't looked closely, but can you optimize the case where there is no dali server configured?  I.e. this ESP is configured to be completely detached from DALI.